### PR TITLE
Create New Account without Lock.js

### PR DESCRIPTION
**Pull request by djones**
Currently when a user is logged in and they want to add an email/pass identity, the user has to fill out a Lock.js dialog like this:

screen shot 2018-01-16 at 3 29 05 pm

This works but isn't ideal because they have to enter their name and email address in when we already know what those are. There's also opportunity for them to enter in a different email address to the one stored on their profile.

This PR is the `percy-web` portion of fixing that. This PR introduces a new form which allows you to simply set a password and be done. No lock.js dialog required. It looks like this:

screen shot 2018-01-23 at 10 40 06 am

Sister PR here https://github.com/percy/percy-api/pull/341

### CODE REVIEW

**Comment by cadeParade** [comment]
Just thinking .... What happens when a user requests a password change, doesn't complete it, changes their email, and completes that verification. Does this scenario work?

**Comment by djones** [comment]
@cadeParade good thought but that's somewhat outside the scope of this PR. This PR is solely focused on creating a new account where there is no verification needed.

File: app/components/forms/account-new.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          type: 'POST',
+          url: utils.buildApiUrl('identities'),
+          data: {data: {attributes: {password: desiredPassword}}},
+        })
+          .done(() => {
+            this.set('isSaveSuccessful', true);
+            this.sendAction('saveSuccess');
+          })
+          .fail(response => {
+            let errorData = response.responseJSON['errors'];
```
Comment (djones) [inline app/components/forms/account-new.js]: I might add a check here to see if there are errors first as that may not always be the case in the response.

File: app/validations/account-new.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import {validateLength, validateFormat} from 'ember-changeset-validations/validators';
+
+export default {
+  password: [
```
Comment (djones) [inline app/validations/account-new.js]: This matches our settings in Auth0

File: app/templates/settings/connected-accounts.hbs [hunk 3]
```diff
@@ -1,10 +1 @@
...
-  <h2 class="f1 text-weight-regular">Connected Accounts</h2>
-
-  {{connected-accounts-panel
-    identities=model
-    deleteIdentity=(route-action "deleteIdentity")
-    addIdentity=(route-action "addIdentity")
-  }}
-
-</div>
+{{connected-accounts-list model=model}}
```
Comment (cadeParade) [inline app/templates/settings/connected-accounts.hbs]: I would recommend renaming `model` to something more descriptive as soon as possible (which is here).
Previously it was `identities` so I am guessing it is identities here?
so something like 
`{{connected-accounts-list identities=model}}`.
Then in the `connected-accounts-list` component, rename `model` to `identities` and everyone knows what we're talking about 👍

File: tests/acceptance/connected-accounts-test.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import setupAcceptance, {setupSession} from '../helpers/setup-acceptance';
```
Comment (cadeParade) [inline tests/acceptance/connected-accounts-test.js]: Optionally convert to path beginning with `percy-web`

File: tests/acceptance/profile-test.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import setupAcceptance, {setupSession} from '../helpers/setup-acceptance';
-import UserSettingsPageObject from '../pages/user-settings';
+import UserSettingsPageObject from '../pages/user-settings-page';
```
Comment (cadeParade) [inline tests/acceptance/profile-test.js]: thanks for fixing this 😬

File: app/components/connected-accounts-list.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  store: service(),
+  showCreateAccount: false,
+  actions: {
+    accountCreated() {
+      this.set('showCreateAccount', false);
+
+      this.get('flashMessages').success('You may now login using this account.', {
+        title: 'Email/password account created!',
+      });
+      this.get('model').reload();
```
Comment (cadeParade) [inline app/components/connected-accounts-list.js]: Renaming this to `identities` would make it clearer what this is reloading. Model is so vague 😢

Comment (cadeParade) [reply app/components/connected-accounts-list.js]: (related comment below)

File: app/components/forms/account-new.js (hunk 1 above)
Comment (cadeParade) [reply app/components/forms/account-new.js]: Yes that's a great idea

File: app/components/connected-accounts-list.js (hunk 6 above)
Comment (djones) [reply app/components/connected-accounts-list.js]: Fixed with de08fabf2a713f637a63bf6fefeb4dd96dcfc67f

File: app/components/forms/account-new.js (hunk 1 above)
Comment (djones) [reply app/components/forms/account-new.js]: Fixed and improved testing with 1363818314fd583bc276df39a1ac0686dc79a7c4

File: tests/acceptance/connected-accounts-test.js (hunk 4 above)
Comment (djones) [reply tests/acceptance/connected-accounts-test.js]: Fixed with db0c815a70fe51a7862be7b087ae795833571ff5

**Pull request merged**